sdk/core/core/inc/az_span_span.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_SPAN_SPAN_H
+#define AZ_SPAN_SPAN_H
+
+#include <az_mut_span.h>
+#include <az_span.h>
+
```

Comment:
You don't need az_mut_span.h - it looks like you're not using it,
#include <az_result.h> - because you use az_result

#include <stddef.h> - because you're using size_t